### PR TITLE
Update `CONTAINERS_COUNT` environment variable if sidecar container was added by another mutating webhook

### PR DESCRIPTION
**Pull request by mjgrzybek**
# Description
Currently `CONTAINERS_COUNT` environment variable is set once when initCointainer with OneAgent installer is created.
If there are more mutating webhooks, e.g. istio, additional sidecar containers are not taken into account.

This change adds support for such scenario by setting correct `COINTAINERS_COUNT` in webook reinvoke phase.

## How can this be tested?
1. setup dybakube with applicationMonitoring
2. install istio
3. create two namespaces, one istio-injected, one not
4. deploy pod in both namespaces and check if `CONTAINERS_COUNT` is set to 2 and 1 respectively

## Checklist
- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Review by 0sewa0 — COMMENTED** [review]
LGTM, added the `GetPodName` to make it more consistent and hopefully less confusing

File: src/kubeobjects/env.go [hunk 1]
```diff
@@ -21,6 +23,16 @@ func EnvVarIsIn(envVars []corev1.EnvVar, envVarToCheck string) bool {
 	return false
 }
 
+func AddOrUpdate(envvars []corev1.EnvVar, desiredEnvVar corev1.EnvVar) []corev1.EnvVar {
```
[the remark this answers is not in the document]
Comment (chrismuellner) [reply src/kubeobjects/env.go:26]: maybe update `envVarList` in other places in this package

File: src/kubeobjects/pod.go [hunk 2]
```diff
@@ -17,3 +17,13 @@ func GetPod(ctx context.Context, clt client.Reader, name, namespace string) (*co
...
 }
+
+// GetPodName returns the name of the pod.
+// During the webhook injection the pod.Name is not always set yet, in which case it returns the pod.GeneraName
+func GetPodName(pod corev1.Pod) string {
+	name := pod.GenerateName
+	if pod.Name != "" {
+		name = pod.Name
+	}
+	return name
```
Comment (chrismuellner) [inline src/kubeobjects/pod.go:28]:
```suggestion
	if pod.Name != "" {
		return pod.Name
	}
	return pod.GenerateName
```

File: src/webhook/mutation/pod_mutator/oneagent_mutation/containers.go [hunk 3]
```diff
@@ -24,7 +32,7 @@ func (mutator *OneAgentPodMutator) mutateUserContainers(request *dtwebhook.Mutat
 // that doesn't conflict with the previous environment variables of the originally injected containers
 func (mutator *OneAgentPodMutator) reinvokeUserContainers(request *dtwebhook.ReinvocationRequest) bool {
 	pod := request.Pod
-	initContainer := findOneAgentInstallContainer(pod.Spec.InitContainers)
+	oneAgentInstallContainer := findOneAgentInstallContainer(pod.Spec.InitContainers)
```
Comment (chrismuellner) [inline src/webhook/mutation/pod_mutator/oneagent_mutation/containers.go:35]: is `oneAgentInstallContainer` more explicit than just `initContainer`?

File: src/webhook/mutation/pod_mutator/oneagent_mutation/mutator.go [hunk 4]
```diff
@@ -57,7 +58,7 @@ func (mutator *OneAgentPodMutator) Reinvoke(request *dtwebhook.ReinvocationReque
 	if !mutator.Injected(request.BaseRequest) {
 		return false
 	}
-	log.Info("reinvoking", "podName", request.Pod.GenerateName)
+	log.Info("reinvoking", "pod.Name", request.PodName())
```
Comment (chrismuellner) [inline src/webhook/mutation/pod_mutator/oneagent_mutation/mutator.go:61]: inconsistent log key

File: src/webhook/mutation/pod_mutator/pod_mutator.go [hunk 5]
```diff
@@ -69,7 +68,7 @@ func (webhook *podMutatorWebhook) Handle(ctx context.Context, request admission.
 			webhook.recorder.sendPodUpdateEvent()
 			return createResponseForPod(mutationRequest.Pod, request)
 		}
-		log.Info("pod already injected, no change", "podName", podName)
+		log.Info("no change to pod injection", "podName", podName)
```
Comment (chrismuellner) [inline src/webhook/mutation/pod_mutator/pod_mutator.go:71]: weirdly phrased

**Review by chrismuellner — CHANGES_REQUESTED** [review]

**Review by chrismuellner — APPROVED** [review]

**Pull request merged**